ui/pad-coms.ts

```diff
@@ -1,37 +1,46 @@
 import { SMX_USB_PRODUCT_ID, SMX_USB_VENDOR_ID, SMXStage } from "../sdk";
-import { stages$, nextStatusTextLine$, statusText$, uiState } from "./state";
+import { stages$, nextStatusTextLine$, uiState, selectedStageSerial$ } from "./state";
 
 export async function promptSelectDevice() {
   const devices = await navigator.hid.requestDevice({
     filters: [{ vendorId: SMX_USB_VENDOR_ID, productId: SMX_USB_PRODUCT_ID }],
   });
 
   if (!devices.length || !devices[0]) {
-    uiState.set(statusText$, "no device selected");
+    uiState.set(nextStatusTextLine$, "no device selected in prompt");
     return;
   }
 
-  await open_smx_device(devices[0]);
+  await open_smx_device(devices[0], true);
 }
 
-export async function open_smx_device(dev: HIDDevice) {
+export async function open_smx_device(dev: HIDDevice, autoSelect = false) {
   if (!dev.opened) {
     await dev.open();
   }
 
   const stage = new SMXStage(dev);
-  await stage.init().then(() => {
-    uiState.set(stages$, (stages) => {
-      return {
-        ...stages,
-        [stage.info?.player || 0]: stage, // TODO: Is there a better way to handle this?
-      };
-    });
-    uiState.set(
-      nextStatusTextLine$,
-      `status: device opened: ${dev.productName}:P${stage.info?.player}:${stage.info?.serial}`,
-    );
+  await stage.init();
+
+  const serial = stage.info?.serial;
+  if (!serial) {
+    uiState.set(nextStatusTextLine$, "selected pad failed to report a serial number");
+    return;
+  }
+
+  uiState.set(stages$, (stages) => {
+    return {
+      ...stages,
+      [serial]: stage, // TODO: Is there a better way to handle this?
```

Comment:
Can remove this TODO. Was for a different situation back when I wrote it. 
```suggestion
      [serial]: stage,
```